src/st_stuff.c

```diff
@@ -718,6 +747,39 @@ void ST_updateWidgets(void)
         keyboxes[i] = keyboxes[i]==-1 || sts_traditional_keys ? i+3 : i+6;
     }
 
+  // [crispy] blinking key or skull in the status bar
+  if (plyr->keyblinktics)
+  {
+    if (st_classicstatusbar || !hud_displayed)
```

Comment:
I believe that `WOOFHUD` allows to hide the keys widget, in which case we should check for that here, so as to avoid playing the sound. How exactly would I check that, though?

Reply:
I can check for it, but it seems to be false only when using the Crispy HUD, or momentarily when changing HUD distributions. In short, it doesn't seem to be useful.

Reply:
I'll have a look later. In general, I am pretty unsure about playing the sound anyway.

Reply:
Wait a minute. You can't have the Boom `w_keys` widget and the status bar visible at the same time, and this code only affects the status bar, so everything should be alright.

Reply:
The drawing aspect is fine. Again, the problem is the sound; only the Status Bar code has the the checks that stop the blinking and the ones that play the sound. I'd guess that such shared behavior should go somewhere else, before the Bar- and HUD-specific code, but where?

Reply:
Yes, right. The problem is that `ST_updateWidgets()` is called during each tic, no matter if the status bar widgets are shown or not.

Reply:
BTW, the sound is also played if Crispy HUD is selected but set to "off".

Reply:
I don't think we need many more checks; just one for the Crispy HUD not being off (`draw_crispy_hud`?), and another for the Boom keys widget being drawn.

Reply:
Let's just pretend that `hud_active > 0` means that the key widget is shown, either in the Boom HUD or Crispy HUD.

Reply:
Done in 229d2b67133f08f5db8b9142fac567df846b9af7.